docs/tutorials/Containers.md

```diff
@@ -0,0 +1,475 @@
+    sbatch \
+      [...]
+      --wrap "singularity exec --bind ${cwd}:${docker_cwd} $IMAGE ${job_shell} ${script}"
+  """
+```
+
+Putting this all together, a complete SLURM + Singularity config might look like this: 
+
+```
+backend {
+  default = slurm
+
+  providers {
+    slurm {
+      actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"                                                                                     
+      config {
+        runtime-attributes = """
+        Int runtime_minutes = 600
+        Int cpus = 2
+        Int requested_memory_mb_per_core = 8000
+        String? docker
+        """
+
+        submit = """
+            sbatch \
+              --wait \
+              -J ${job_name} \
+              -D ${cwd} \
+              -o ${out} \
+              -e ${err} \
+              -t ${runtime_minutes} \
+              ${"-c " + cpus} \
+              --mem-per-cpu=${requested_memory_mb_per_core} \
+              --wrap "/bin/bash ${script}"
+        """
+
+        submit-docker = """
+            # Ensure singularity is loaded if it's installed as a module
+            module load Singularity/3.0.1
+            
+            # Build the Docker image into a singularity image
+            IMAGE=${cwd}/${docker}.sif
+            singularity build $IMAGE docker://${docker}
+
+            # Submit the script to SLURM
+            sbatch \
+              --wait \
+              -J ${job_name} \
+              -D ${cwd} \
+              -o ${cwd}/execution/stdout \
+              -e ${cwd}/execution/stderr \
+              -t ${runtime_minutes} \
+              ${"-c " + cpus} \
+              --mem-per-cpu=${requested_memory_mb_per_core} \
+              --wrap "singularity exec --bind ${cwd}:${docker_cwd} $IMAGE ${job_shell} ${script}"
+        """
+
+        kill = "scancel ${job_id}"
+        check-alive = "squeue -j ${job_id}"
+        job-id-regex = "Submitted batch job (\\d+).*"
+      }
+    }
+  }
+}
+
+```
+
+#### Without Setuid
+In addition, if you or your sysadmins were not able to give `setuid` permissions to `singularity`, you'll have to modify the config further to ensure the use of sandbox images:
+
+```
+submit-docker = """
+    [...]
+
+    # Build the Docker image into a singularity image
+    # We don't add the .sif file extension because sandbox images are directories, not files
+    IMAGE=${cwd}/${docker}
+    singularity build --sandbox $IMAGE docker://${docker}
+
+    # Now submit the job
+    # Note the use of --userns here
+    sbatch \
+      [...]
+      --wrap "singularity exec --userns --bind ${cwd}:${docker_cwd} $IMAGE ${job_shell} ${script}"
+"""
+```
+
+
+#### Singularity Cache
+By default, Singularity will cache the Docker images you pull in `~/.singularity`, in your home directory.
+
+However, if you are sharing your Docker images with other users, you probably want this cache to exist somewhere that is accessible to everyone.
+
```

Comment:
It's also important the variable is exported (not just defined) in the .bashrc or similar otherwise it won't take effect. I'd advise users to do this immediately because even if they *don't* need to share layers, their $HOME is going to fill up almost immediately, they will get locked out, and have to ask their admin for help. Best to avoid that.

Reply:
I've added a more explicit requirement to export the variable, more comments here: https://github.com/broadinstitute/cromwell/pull/4635#discussion_r258272667,